Snhut 4926 auth bugfix

shafi-sadaindia opened the pull request:
### Description
If the signed-out user hits URL: /turbine-admin, the Application will redirect to the same URL after login.

- Fixed redirect URL Globally.

- [JIRA Ticket](https://collectiveshift.atlassian.net/browse/SNHUT-4926)

#### Checklist

- [x] **Assumptions of User Story met (including acceptance criteria)**
- [x] **All new necessary Unit tests were CREATED** to promote code coverage
- [x] **All required unit tests are PASSING**
- [x] The proposed functionality was tested manually locally, and it's working fine.
- [ ] PR was properly reviewed by a team member (at least 1 approval required to merge!)
- [x] Unnecessary console.logs AND/OR comments were removed
- [ ] Feature ok-ed by UX designer (If needed - not for back end components) and Product Owner
- [ ] Ensure configuration and/or build changes documented

mk-sada reviewed (changes requested):
This works, but two issues.
1. You've linked [SNHUT-4305](https://collectiveshift.atlassian.net/browse/SNHUT-4305) to this item, but I'm struggling to see the relationship between what you did and what the ticket is asking you to solve.
2. This touches the `auth.guard.ts`, which is a rather crucial component, and I'd rather not make any change to it before alpha release for a PLA item.

---

If you can help me understand why this change is necessary, we can revisit this and merge it after April 6th.

shafi-sadaindia commented:
> This works, but two issues.
> 
> 1. You've linked [SNHUT-4305](https://collectiveshift.atlassian.net/browse/SNHUT-4305) to this item, but I'm struggling to see the relationship between what you did and what the ticket is asking you to solve.
> 2. This touches the `auth.guard.ts`, which is a rather crucial component, and I'd rather not make any change to it before alpha release for a PLA item.
> 
> If you can help me understand why this change is necessary, we can revisit this and merge it after April 6th.

shafi-sadaindia commented:
> This works, but two issues.
> 
> 1. You've linked [SNHUT-4305](https://collectiveshift.atlassian.net/browse/SNHUT-4305) to this item, but I'm struggling to see the relationship between what you did and what the ticket is asking you to solve.
> 2. This touches the `auth.guard.ts`, which is a rather crucial component, and I'd rather not make any change to it before alpha release for a PLA item.
> 
> If you can help me understand why this change is necessary, we can revisit this and merge it after April 6th.

Hi @mk-sada ,

Added wrong ticket by mistake. Actual ticket is https://collectiveshift.atlassian.net/browse/SNHUT-4926. Updated discription

mk-sada commented:
I verified that the code works, and it does address the bug.
Now that we are not in a crunch, I would like us to clean up the auth/reroute logic some more.
Would you mind addressing my inline comments in `auth.service.ts`?

mk-sada commented on microservices/turbine-admin/apps/toboggan-app/src/app/shared/auth/auth.service.ts:
```diff
@@ -374,4 +361,34 @@ export class AuthService {
+        window.location.href = '/dashboard';
+      }
+    } else {
+      if (url === '/' || url === '') {
```
Let's handle the logic for each non-learner user using switch-case statements instead of nested if-else.
Non-learner users being:
- Admin [test_admin.006@leveluplabs.org]
- assessor [test_assessor.006@leveluplabs.org]
- coach [test_coach.006@leveluplabs.org]
- instructor [test_instructor.006@leveluplabs.org]
- LXE [test_LXE.004@leveluplabs.org]

mk-sada commented on microservices/turbine-admin/apps/toboggan-app/src/app/shared/auth/auth.service.ts:
```diff
@@ -374,4 +361,34 @@ export class AuthService {
+          currentUser.userType === UserTypes.CURRICULAM_DESIGNER ||
+          currentUser.userType === UserTypes.LXE
+        ) {
+          this.router.navigate(['/content-management'], {
```
make sure to `await` the `router.navigate` calls. E.g.:

```TypeScript
await this.router.navigate(['/content-management'], {
```
